build-script.py

```diff
@@ -740,6 +817,18 @@ def main():
     if args.degyb_only:
         sys.exit(0)
 
+    run_code_generation(
+        toolchain=args.toolchain,
+        build_dir=realpath(args.build_dir),
+        multiroot_data_file=args.multiroot_data_file,
+        release=args.release,
+        verbose=args.verbose,
+        swiftsyntaxbuilder_destination=os.path.join(SWIFTSYNTAXBUILDER_DIR, "generated")
+    )
```

Comment:
Sorry if I wasn’t clear in https://github.com/apple/swift-syntax/pull/381#discussion_r876791560. I was saying that we shouldn’t run the `SwiftSyntaxBuilderGeneration` as an implicit step in `build-script.py` because `SwiftSyntaxBuilderGeneration` itself might no longer compile after `SwiftSyntaxBuilder` has been re-generated.

My idea was that we should have a flag `--run-swiftsyntaxbuilder-generation` (or something of the sort) and then have this be like
```suggestion
    if args.run_swiftsyntaxbuilder_generation:
      run_code_generation(
          toolchain=args.toolchain,
          build_dir=realpath(args.build_dir),
          multiroot_data_file=args.multiroot_data_file,
          release=args.release,
          verbose=args.verbose,
          swiftsyntaxbuilder_destination=os.path.join(SWIFTSYNTAXBUILDER_DIR, "generated")
      )
      sys.exit(0)
```

In which case we could also move it up to the top of `main`.

Also: Since `build-script-helper.py` now does so many things, we should probably migrate it to using argparse subcommands. But that’s something for a completely standalone PR.